helios-services/src/main/java/com/spotify/helios/agent/AgentService.java

```diff
@@ -347,10 +348,11 @@ public AgentService(final AgentConfig config, final Environment environment)
   }
 
   private DockerClient createDockerClient(final AgentConfig config) {
-    final DockerClient dockerClient;
-    if (isNullOrEmpty(config.getDockerHost().dockerCertPath())) {
-      dockerClient = new PollingDockerClient(config.getDockerHost().uri());
-    } else {
+    final DefaultDockerClient.Builder builder = DefaultDockerClient.builder()
+            .uri(config.getDockerHost().uri())
+            .connectionPoolSize(config.getConnectionPoolSize());
```

Comment:
I think it would be better to not call `.connectionPoolSize(int)` at all if the flag is not specified for helios-agent, that way the default value when not specified still resides within the DefaultDockerClient.Builder class as it does today.